fabtests/prov/efa/src/rdm_remote_exit_early.c

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2025, Amazon.com, Inc.  All rights reserved.
+ *
+ * This software is available to you under the BSD license
+ * below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * This test check the error handling path for receiver when sender exit early
+ * in the middle of send.
+ */
+
+#include <getopt.h>
+#include <shared.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <rdma/fi_tagged.h>
+
+static bool post_rx = false;
+static bool post_tagged = false;
+static bool post_writedata = false;
+
+enum {
+	LONG_OPT_POST_RX,
+};
+
+static int run()
+{
+	int ret;
+
+	ret = ft_init_fabric();
+	if (ret) {
+		FT_PRINTERR("ft_init_fabric", -ret);
+		return ret;
+	}
+
+	/*
+	 * The handshake procedure between server and client will happen in
+	 * either ft_exchange_keys() or ft_sync()
+	 */
+	if (post_writedata) {
+		ret = ft_exchange_keys(&remote);
```

Comment:
Notice: `ft_exchange_keys` finally calls `ft_sync()`, which will call `ft_sync_inband(true)`, so it will cause the receiver to always post an outstanding recv buffer. This is ok for RMA test because the rx buffer post shouldn't matter here, unless we will support `FI_RX_CQ_DATA`